src/lib/utils.js

```diff
@@ -1,8 +1,19 @@
-export function mergeTheme(theme, defaultTheme) {
+import * as defaultTheme from "./style/theme";
+
+function mergeTheme(theme, defaultTheme) {
   return theme && theme.brand
     ? {
         ...defaultTheme.brand,
         ...theme.brand
       }
     : defaultTheme.brand;
 }
+
+const getKey = (key, props) => typeof key === 'function' ? key(props) : key;
+
+export const getThemeProp = (key, defaultKey = null) => props => {
+  if (defaultKey && defaultKey(props)) return defaultKey(props)
```

Comment:
IMO, we don't need to manage "defaultKey" here. Because we don't need to call `getThemeProp ` to return the `default` value ;)